cmd/maestro/server/grpc_server.go

```diff
@@ -150,8 +176,19 @@ func (svr *GRPCServer) Publish(ctx context.Context, pubReq *pbv1.PublishRequest)
 			return nil, fmt.Errorf("failed to update resource: %v", err)
 		}
 	case common.DeleteRequestAction:
-		err := svr.resourceService.MarkAsDeleting(ctx, res.ID)
+		res, err = decode(eventType.CloudEventsDataType, evt)
 		if err != nil {
+			return nil, fmt.Errorf("failed to decode cloudevent: %v", err)
+		}
+		var deletionTimestamp time.Time
+		evtExtensions := evt.Context.GetExtensions()
+		if _, ok := evtExtensions[codec.ExtensionWorkMeta]; ok {
+			if _, exists := evtExtensions[types.ExtensionDeletionTimestamp]; exists {
+				deletionTimestamp = time.Now()
```

Comment:
we are not setting 
```go
 res.Meta.DeletedAt = gorm.DeletedAt{
	Time:  deletionTimestamp,
        Valid: true,
}
```

just like what we do for create to ensure that the two  field values (the timestamp in metadata and in resource bundle) are the same.

Reply:
Please see https://github.com/openshift-online/maestro/pull/159/files#r1673551640

Reply:
I encountered to apply the `deleteAt` for resource using `Update`, let me try again when we have `Valid: true`.

Reply:
Thank you for looking. Please note, this should be done in the line comment at https://github.com/openshift-online/maestro/pull/159#discussion_r1673551640

Reply:
Looks like the `valid: true` work.
I updated this so that work meta and resource `deleteAt` are using the same `deletionTimestamp`.